charts/super-agent/charts/super-agent-deployment/templates/_helpers.tpl

```diff
@@ -283,3 +283,116 @@ value is provided, it defaults to `""` (empty string) so this helper can be used
+*/ -}}
+{{- define "newrelic-super-agent.config.endpoints.systemIdentityCreation" -}}
+{{- $region := include "newrelic.common.region" . -}}
+
+{{- if eq $region "Staging" -}}
+  https://ng-iam-service.staging-service.nr-ops.net/system-identity/graphql
+{{- else if eq $region "EU" -}}
+  https://ng-iam-service.service.eu.nr-ops.net/system-identity/graphql
+{{- else if eq $region "US" -}}
+  https://ng-iam-service.service.nr-ops.net/system-identity/graphql
+{{- else if eq $region "Local" -}}
+  {{- /* Accessing the value directly without protection. A developer should now how to read the error. */ -}}
+  {{ .Values.development.backend.systemIdentityCreation }}
+{{- else -}}
+  {{- fail "Unknown/unsupported region set for this chart" -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the name key for the ClientId Key inside the secret.
+*/}}
+{{- define "newrelic-super-agent.auth.l1IdentityCredentialsKey.clientIdKeyName" -}}
+{{- include "newrelic-super-agent.auth.identityCredentialsL1._customClientIdKey" . | default "clientIdKey" -}}
+{{- end -}}
+
+{{/*
+Return the name key for the ClientSecret Key inside the secret.
+*/}}
+{{- define "newrelic-super-agent.auth.l1IdentityCredentialsKey.clientSecretKeyName" -}}
+{{- include "newrelic-super-agent.auth.identityCredentialsL1._customClientSecretKey" . | default "clientSecretKey" -}}
+{{- end -}}
+
+{{/*
+Return the name of the secret holding the clientdId and ClientSecret
+*/}}
+{{- define "newrelic-super-agent.auth.customl1IdentitySecretName" -}}
+{{- if .Values.customL1IdentitySecretName -}}
+  {{- .Values.customL1IdentitySecretName -}}
+{{- else if .Values.global -}}
+  {{- if .Values.global.customL1IdentitySecretName -}}
+    {{- .Values.global.customL1IdentitySecretName -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the name key for the ClientID inside the secret.
+*/}}
+{{- define "newrelic-super-agent.auth.identityCredentialsL1._customClientIdKey" -}}
+{{- if .Values.customL1IdentityClientIdSecretKey -}}
+  {{- .Values.customL1IdentityClientIdSecretKey -}}
+{{- else if .Values.global -}}
+  {{- if .Values.global.customL1IdentityClientIdSecretKey }}
+    {{- .Values.global.customL1IdentityClientIdSecretKey -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the name key for the ClientSecret inside the secret.
+*/}}
+{{- define "newrelic-super-agent.auth.identityCredentialsL1._customClientSecretKey" -}}
+{{- if .Values.customL1IdentityClientSecretSecretKey -}}
+  {{- .Values.customL1IdentityClientSecretSecretKey -}}
+{{- else if .Values.global -}}
+  {{- if .Values.global.customL1IdentityClientSecretSecretKey }}
+    {{- .Values.global.customL1IdentityClientSecretSecretKey -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{- define "newrelic-super-agent.auth.generatedIdentityCredentialsSecretName" -}}
+{{ include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "preinstall-client-credentials" ) }}
+{{- end -}}
```

Comment:
nit this is used by the function below, I would place it just below that

Reply:
I dedicated this commit to you 😗  c3cdee864ca2b501b5f1b945abdb8d2b90be10cb